localstack-core/localstack/services/apigateway/legacy/provider.py

```diff
@@ -987,11 +987,81 @@ def get_method_response(
+
+        error_messages = []
+        for index, operation in enumerate(patch_operations):
+            op = operation.get("op")
+            if op not in VALID_PATCH_OPERATIONS:
+                error_messages.append(
+                    f"Value '{op}' at 'updateMethodResponseInput.patchOperations.{index + 1}.member.op' "
+                    f"failed to satisfy constraint: Member must satisfy enum value set: [{', '.join(VALID_PATCH_OPERATIONS)}]"
+                )
+
+        if not re.fullmatch(r"[1-5]\d\d", status_code):
+            error_messages.append(
+                f"Value '{status_code}' at 'statusCode' failed to satisfy constraint: "
+                "Member must satisfy regular expression pattern: [1-5]\\d\\d"
+            )
+
+        if error_messages:
+            prefix = f"{len(error_messages)} validation error{'s' if len(error_messages) > 1 else ''} detected: "
+            raise CommonServiceException(
+                code="ValidationException",
+                message=prefix + "; ".join(error_messages),
+            )
+
+        moto_rest_api = get_moto_rest_api(context, rest_api_id)
+        moto_resource = moto_rest_api.resources.get(resource_id)
+        if not moto_resource:
+            raise NotFoundException("Invalid Resource identifier specified")
+
+        moto_method = moto_resource.resource_methods.get(http_method)
+        if not moto_method:
+            raise NotFoundException("Invalid Method identifier specified")
+
+        method_response = moto_method.method_responses.get(status_code)
+        if not method_response:
+            raise NotFoundException("Invalid Response status code specified")
+
+        for patch_operation in patch_operations:
+            op = patch_operation["op"]
+            path = patch_operation["path"]
+            value = patch_operation.get("value")
+
+            if path.startswith("/responseParameters/"):
+                param_name = path.removeprefix("/responseParameters/")
+                if op in ("add", "replace"):
+                    method_response.response_parameters[param_name] = value == "true"
+                elif op == "remove":
+                    method_response.response_parameters.pop(param_name)
+
+            elif path.startswith("/responseModels/"):
+                param_name = path.removeprefix("/responseModels/")
+                param_name = param_name.replace("~1", "/")
+                if op in ("add", "replace"):
+                    method_response.response_models[param_name] = value
+                elif op == "remove":
+                    method_response.response_models.pop(param_name)
+
+        response: MethodResponse = method_response.to_json()
+
+        # AWS doesn't send back empty responseParameters or responseModels
+        if method_response.response_parameters == {}:
+            response.pop("responseParameters")
+        if method_response.response_models == {}:
+            response.pop("responseModels")
```

Comment:
Very nice attention to detail! 👌 

nit: doing `if not method.response.response_parameters` is a bit more Pythonic, because an empty dictionary is falsy.